File: app/src/main/java/com/BlizzardArmory/overwatch/Competitive/Average.java

package com.BlizzardArmory.overwatch.Competitive;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Average {

    @SerializedName("allDamageDoneAvgPer10Min")
    @Expose
    private int allDamageDoneAvgPer10Min;
    @SerializedName("deathsAvgPer10Min")
    @Expose
    private float deathsAvgPer10Min;
    @SerializedName("eliminationsAvgPer10Min")
    @Expose
    private float eliminationsAvgPer10Min;
    @SerializedName("finalBlowsAvgPer10Min")
    @Expose
    private float finalBlowsAvgPer10Min;
    @SerializedName("healingDoneAvgPer10Min")
    @Expose
    private int healingDoneAvgPer10Min;
    @SerializedName("objectiveKillsAvgPer10Min")
    @Expose
    private float objectiveKillsAvgPer10Min;
    @SerializedName("objectiveTimeAvgPer10Min")
    @Expose
    private String objectiveTimeAvgPer10Min;
    @SerializedName("soloKillsAvgPer10Min")
    @Expose
    private float soloKillsAvgPer10Min;
    @SerializedName("timeSpentOnFireAvgPer10Min")
    @Expose
    private String timeSpentOnFireAvgPer10Min;

    public int getAllDamageDoneAvgPer10Min() {
        return allDamageDoneAvgPer10Min;
    }

    public void setAllDamageDoneAvgPer10Min(int allDamageDoneAvgPer10Min) {
        this.allDamageDoneAvgPer10Min = allDamageDoneAvgPer10Min;
    }

    public float getDeathsAvgPer10Min() {
        return deathsAvgPer10Min;
    }

    public void setDeathsAvgPer10Min(float deathsAvgPer10Min) {
        this.deathsAvgPer10Min = deathsAvgPer10Min;
    }

    public float getEliminationsAvgPer10Min() {
        return eliminationsAvgPer10Min;
    }

    public void setEliminationsAvgPer10Min(float eliminationsAvgPer10Min) {
        this.eliminationsAvgPer10Min = eliminationsAvgPer10Min;
    }

    public float getFinalBlowsAvgPer10Min() {
        return finalBlowsAvgPer10Min;
    }

    public void setFinalBlowsAvgPer10Min(float finalBlowsAvgPer10Min) {
        this.finalBlowsAvgPer10Min = finalBlowsAvgPer10Min;
    }

    public int getHealingDoneAvgPer10Min() {
        return healingDoneAvgPer10Min;
    }

    public void setHealingDoneAvgPer10Min(int healingDoneAvgPer10Min) {
        this.healingDoneAvgPer10Min = healingDoneAvgPer10Min;
    }

    public float getObjectiveKillsAvgPer10Min() {
        return objectiveKillsAvgPer10Min;
    }

    public void setObjectiveKillsAvgPer10Min(float objectiveKillsAvgPer10Min) {
        this.objectiveKillsAvgPer10Min = objectiveKillsAvgPer10Min;
    }

    public String getObjectiveTimeAvgPer10Min() {
        return objectiveTimeAvgPer10Min;
    }

    public void setObjectiveTimeAvgPer10Min(String objectiveTimeAvgPer10Min) {
        this.objectiveTimeAvgPer10Min = objectiveTimeAvgPer10Min;
    }

    public float getSoloKillsAvgPer10Min() {
        return soloKillsAvgPer10Min;
    }

    public void setSoloKillsAvgPer10Min(float soloKillsAvgPer10Min) {
        this.soloKillsAvgPer10Min = soloKillsAvgPer10Min;
    }

    public String getTimeSpentOnFireAvgPer10Min() {
        return timeSpentOnFireAvgPer10Min;
    }

    public void setTimeSpentOnFireAvgPer10Min(String timeSpentOnFireAvgPer10Min) {
        this.timeSpentOnFireAvgPer10Min = timeSpentOnFireAvgPer10Min;
    }

}

File: app/src/main/java/com/BlizzardArmory/model/overwatch/Average.kt
package com.BlizzardArmory.model.overwatch

import com.google.gson.annotations.SerializedName


import java.text.DecimalFormat
import java.text.NumberFormat
import java.util.*

/**
 * The type Average.
 */

data class Average(

    @SerializedName("allDamageDoneAvgPer10Min")
    var allDamageDoneAvgPer10Min: Double,

    @SerializedName("barrierDamageDoneAvgPer10Min")
    var barrierDamageDoneAvgPer10Min: Double,

    @SerializedName("deathsAvgPer10Min")
    var deathsAvgPer10Min: Float,

    @SerializedName("eliminationsAvgPer10Min")
    var eliminationsAvgPer10Min: Float,

    @SerializedName("finalBlowsAvgPer10Min")
    var finalBlowsAvgPer10Min: Float,

    @SerializedName("healingDoneAvgPer10Min")
    var healingDoneAvgPer10Min: Double,

    @SerializedName("heroDamageDoneAvgPer10Min")
    var heroDamageDoneAvgPer10Min: Double,

    @SerializedName("objectiveKillsAvgPer10Min")
    var objectiveKillsAvgPer10Min: Float,

    @SerializedName("objectiveTimeAvgPer10Min")
    var objectiveTimeAvgPer10Min: String,

    @SerializedName("soloKillsAvgPer10Min")
    var soloKillsAvgPer10Min: Float,

    @SerializedName("timeSpentOnFireAvgPer10Min")
    var timeSpentOnFireAvgPer10Min: String
) {
    val average: HashMap<String, String>
        get() {
            val averageList = HashMap<String, String>()
            val formatter: NumberFormat = DecimalFormat("#0")
            averageList["ALL DAMAGE DONE - AVG PER 10 MIN"] = formatter.format(allDamageDoneAvgPer10Min)
            averageList["BARRIER DAMAGE DONE - AVR PER 10 MIN"] = formatter.format(barrierDamageDoneAvgPer10Min)
            averageList["DEATHS - AVG PER 10 MIN"] = deathsAvgPer10Min.toString()
            averageList["ELEMINATIONS - AVG PER 10 MIN"] = eliminationsAvgPer10Min.toString()
            averageList["FINAL BLOWS - AVG PER 10 MIN"] = finalBlowsAvgPer10Min.toString()
            averageList["HEALING DONE - AVG PER 10 MIN"] = formatter.format(healingDoneAvgPer10Min)
            averageList["HERO DAMAGE DONE - AVG PER 10 MIN"] = formatter.format(heroDamageDoneAvgPer10Min)
            averageList["OBJECTIVE KILLS - AVG PER 10 MIN"] = objectiveKillsAvgPer10Min.toString()
            averageList["OBJECTIVE TIME - AVG PER 10 MIN"] = objectiveTimeAvgPer10Min
            averageList["SOLO KILLS - AVG PER 10 MIN"] = soloKillsAvgPer10Min.toString()
            averageList["TIME SPENT ON FIRE - AVG PER 10 MIN"] = timeSpentOnFireAvgPer10Min
            return averageList
        }
}